src/core/common/message.cpp

```diff
@@ -110,9 +127,13 @@ ThreadError MessagePool::FreeBuffers(Buffer *aBuffer)
     while (aBuffer != NULL)
     {
         tmpBuffer = aBuffer->GetNextBuffer();
+#if OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
+        otPlatMessagePoolFree(static_cast<struct BufferHeader *>(aBuffer));
+#else // OPENTHREAD_CONFIG_PLATFORM_MESSAGE_MANAGEMENT
         aBuffer->SetNextBuffer(mFreeBuffers);
         mFreeBuffers = aBuffer;
         mNumFreeBuffers++;
```

Comment:
Contrary to your claim, my platform implementation does continue to use mNumFreeBuffers and mFreeBuffers.  It is useful in that it continues to allow me to configure the number of objects using the existing configuration for OPENTHREAD_CONFIG_NUM_MESSAGE_BUFFERS. I specifically pass mNumFreeBuffers as a pointer to allow ReclaimBuffers to continue to work as designed.  It is the platform's job to update the mNumFreeBuffers value as buffers are allocated and freed.